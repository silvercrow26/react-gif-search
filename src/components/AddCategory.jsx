import PropTypes from "prop-types";
import { useState } from "react"



export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('');

    const onInputChange = ({target}) => {
        setinputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim();

        if(newInputValue.length <= 1) return setinputValue('');
            // onNewCategory(cat => [inputValue, ...cat])
            onNewCategory(newInputValue);
            setinputValue('');      
    }

  return (
    <form aria-label="form" onSubmit={onSubmit}>
        <input
            type="text"
            placeholder="Buscar gif"
            value={ inputValue }
            onChange={onInputChange}
        />
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}