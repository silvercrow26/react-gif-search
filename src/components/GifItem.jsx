import PropTypes from "prop-types";

export const GifItem = ({url, title}) => {

  return (
    <div className='card'>
        <img data-testid="hola" src={url} alt={ title }></img>
        {/* <p>{title}</p> */}
    </div>
  )

};

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};