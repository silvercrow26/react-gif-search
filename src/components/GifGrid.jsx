
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";


//Nunca poner la ejecucion de una funcion directamente en un componente
export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs(category);

    return (
    <>

        <h4>{ category }</h4>
        {         
            isLoading && (<h4 >Cargando gifs...</h4>)
        }
 
        <div className="card-grid">
                {images.map( (image) => (
                   <GifItem key={image.id} 
                   { ...image }
                   />
                ))
                }
        </div><br></br>
    </>
  )

}
