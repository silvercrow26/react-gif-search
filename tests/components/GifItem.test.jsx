import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Testing a GifItem', () => { 

    const title = 'Kirino'
    const url = 'https://kirino.com/kirino.jpg'


     test('Debe hacer match con el snapshot', () => { 

        const {container} = render(<GifItem url={title} title={url}/>)
        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 

        render(<GifItem url={url} title={title}/>)
        
        //Desestructurando propiedades de img
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
      
        // screen.debug();
        // // console.log(screen.getByRole('img').src)
        // expect(screen.getByRole('img').src).toBe(url);
        // expect((screen.getByTestId('hola').alt)).toBe(title);
    });

    // test('Debe de mostrar el tÃ­tulo en el componente', () => { 

    //     render(<GifItem url={url} title={title}/>)

    //     expect(screen.getByText(title)).toBeTruthy();
    // })
})