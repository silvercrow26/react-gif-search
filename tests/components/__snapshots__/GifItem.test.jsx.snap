// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing a GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="https://kirino.com/kirino.jpg"
      data-testid="hola"
      src="Kirino"
    />
  </div>
</div>
`;
